pkgs/check-spec-patches.sh

```diff
@@ -0,0 +1,89 @@
+#!/bin/bash
+# check-spec-patches.sh
+#	usage:
+#		check-spec-patches.sh <specfile>
+#	specfile: valid specfile.
+#
+# Description:
+#    This script checks that:
+#      a) Specfile applies patches that were specified by `PatchN:`
+#         i.e. for every PatchN there is %patchN
+#      b) Each patch exists in the current directory
+#         i.e. For every PatchN: <file.patch> there is <file.patch> in the current directory.
+#      c) Each Patch in directory is tracked in git
+#         i.e. Every <file.patch> is commited in git
+#    If any of the rules is broken then the script throws an error.
+#
+# Dependencies: Ruby interpreter, bash
+
+function die {
+	for var in "$@"
+	do
+		echo "$var" 1>&2
+	done
+	exit 1
+}
+
+# Accepts single argument with the header content.
+function format_header {
+	printf "==== %s ====" "${1}"
+}
+
+file="$1"
+
+if [ -z "$file" ]; then
+	die "No specfile supplied"
+fi
+
+sources=""
+applied=""
+existing=""
+
+# Get patches that are packaged
+sources=$(grep -e "^Patch[0-9]*:" $file)
```

Comment:
This is better handled by spectool, isn't it?

Reply:
I wasn't aware of such a tool, it seems to do the job in a parseable way.
But it fails with fedora/rawhide ruby for me (though fedpkg doesn't complain and can create srpm)...
```console
$ spectool --patches ./ruby.spec 
error: ./ruby.spec: line 112: failed to load macro file /home/jprokop/rpmbuild/SOURCES/macros.ruby
RPM Failed to parse spec file.
```